cluster-provision/k8s/1.25/provision.sh

```diff
@@ -48,11 +60,43 @@ function pull_container_retry() {
     fi
 }
 
-# Install modules of the initrd kernel
-dnf install -y "kernel-modules-$(uname -r)"
+export CRIO_VERSION=1.25
+cat << EOF >/etc/yum.repos.d/devel_kubic_libcontainers_stable.repo
+[devel_kubic_libcontainers_stable]
+name=Stable Releases of Upstream github.com/containers packages (CentOS_8_Stream)
+type=rpm-md
+baseurl=https://storage.googleapis.com/kubevirtci-crio-mirror/devel_kubic_libcontainers_stable/
+gpgcheck=0
+enabled=1
+EOF
+cat << EOF >/etc/yum.repos.d/devel_kubic_libcontainers_stable_cri-o_${CRIO_VERSION}.repo
+[devel_kubic_libcontainers_stable_cri-o_${CRIO_VERSION}]
+name=devel:kubic:libcontainers:stable:cri-o:${CRIO_VERSION} (CentOS_8_Stream)
+type=rpm-md
+baseurl=https://storage.googleapis.com/kubevirtci-crio-mirror/devel_kubic_libcontainers_stable_cri-o_${CRIO_VERSION}
+gpgcheck=0
+enabled=1
+EOF
+
+packages_version=$(getKubernetesClosestStableVersion)
+# Add Kubernetes release repository.
+# use repodata from GCS bucket, since the release repo might not have it right after the release
+# we deduce the https path from the gcs path gs://kubernetes-release/release/${version}/rpm/x86_64/
+# see https://github.com/kubernetes/kubeadm/blob/main/docs/testing-pre-releases.md#availability-of-pre-compiled-release-artifacts
+cat <<EOF >/etc/yum.repos.d/kubernetes.repo
+[kubernetes]
+name=Kubernetes Release
+baseurl=https://storage.googleapis.com/kubernetes-release/release/v${packages_version}/rpm/x86_64/
+enabled=1
+gpgcheck=0
+repo_gpgcheck=0
+EOF
+
+dnf install -y centos-release-nfv-openvswitch
+
+dnf install --disableexcludes=kubernetes -y "kernel-modules-$(uname -r)" cloud-utils-growpart patch lvm2 iscsi-initiator-utils iproute-tc cri-o podman libseccomp-devel openvswitch2.16 kubectl-${packages_version} kubeadm-${packages_version} kubelet-${packages_version} kubernetes-cni
```

Comment:
I would prefer please to keep it separated to groups i.e
k8s
openvswitch 
It allows maintaining and see what happens easier, 
it can still be one next to each other in one place

nit: if desired - some of them might better to install as late as possible, so if stuck breaks before, during development no need to wait for the rest

about the repo files - maybe export the files to a folder and just copy them instead in-place ?
better readability


the general ones can be bundled
better with one on each line using same dnf command but with back slash at the end of the line if possible please

Reply:
> 1. I would prefer please to keep it separated to groups i.e
>    `k8s`
>    `openvswitch`
>    It allows maintaining and see what happens easier,
>    it can still be one next to each other in one place

I am not sure how this will make it easier - if there is an issue installing a package you will still have to check node logs to see what has happened. 

> 
> * The general ones can be bundled
> * Maybe there is a way to split to lines, use the same dnf command, but have some separator between groups ?
> 
> 2. Better with one on each line using same dnf command but with back slash at the end of the line if possible please

Yes definitely better for readability - will make the update for this.

> 3. About the repo files - maybe export the files to a folder and just copy them instead in-place ?
>    better readability (can be on other PR of course)

That could be something we could look at in another PR

> 4. 1.24 isn't done by purpose / other code / EOL ?

I think there were slight differences in the 1.24 provider so I decided to leave it as is and let the provider age out. I can take a look again at applying these changes there. 

> 
> Thanks

Thanks Or

Reply:
> > 1. I would prefer please to keep it separated to groups i.e
> >    `k8s`
> >    `openvswitch`
> >    It allows maintaining and see what happens easier,
> >    it can still be one next to each other in one place
> 
> I am not sure how this will make it easier - if there is an issue installing a package you will still have to check node logs to see what has happened.

but it is easier for the eye to catch the section and what belong to that failure, they are separate snippets
and it make it more friendly and welcoming imo in general to read the list